src/vs/code/electron-main/windows.ts

```diff
@@ -761,6 +768,8 @@ export class WindowsManager implements IWindowsMainService {
 			}
 		}
 
+		this.backupService.registerWindowForBackups(vscodeWindow.id, !configuration.workspacePath, configuration.backupFolder, configuration.workspacePath);
```

Comment:
Yes, moved it to `registerWindowForBackups` so it was handled by the `BackupMainService`. Now that you mentioned it though, that was there for a good reason, thanks for pointing that out :+1: